src/Controllers/Frontend.php

```diff
@@ -70,6 +70,32 @@ public function before(Request $request, Application $app)
     }
 
     /**
+     * The default after filter for the controllers in this file.
+     *
+     * Refer to the routing.yml config file for overridding.
+     *
+     * @param Request     $request  The Symfony Request
+     * @param Response    $response The Symfony Request
+     * @param Application $app      The application/container
+     *
+     * @return mixed
+     */
+    public function after(Request $request, Response $response, Application $app)
+    {
+        // Frontend requests should not set cookies. Remove them, to allow Varnish
+        // (or reverse-proxies) to do a better job of caching the request.
+        if (!headers_sent()) {
+            $headersList = headers_list();
+            foreach($headersList as $header) {
+                if (strpos($header, "Set-Cookie: bolt_session=") === 0) {
+                    header_remove("Set-Cookie");
+                }
```

Comment:
It aslo does the [headers_sent()](https://github.com/symfony/HttpFoundation/blob/2.6/Response.php#L332)